source/MaterialXCore/Node.cpp

```diff
@@ -102,16 +102,50 @@ Edge Node::getUpstreamEdge(ConstMaterialPtr material, size_t index) const
     return NULL_EDGE;
 }
 
-OutputPtr Node::getNodeDefOutput(const Edge& edge)
+OutputPtr Node::getNodeDefOutput(ElementPtr connectingElement)
```

Comment:
Only the connecting element of the edge was used, so changed to give this explicitly. Gives more flexible usage of this method, needed below.

Reply:
How much do we worry about breaking backwards compatibility due to signature change.
Can we process the this extra logic outside and produce an edge for testing here for instance?
If not would it be better to add a new method or is the current one just misleading?

Reply:
It's a good point. This method was added by us, in January, but it was never reported as a new addition in the changelong from March, so it "slipped in" under the radar. It's also quite special and not used in general to construct or parse materialx documents. So I doubt it's been used by anyone except us.

The fact that an whole Edge was given before was a mistake, as it's not at all flexible. You might want to know the output  connected to an input without having to construct a full Edge first.

So I think this change is necessary and should be OK.

Reply:
Sounds like we're safe then. Just built-in paranoid about someone chasing us down for API breakages.